modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Albert Gomes Cabral
+ */
+public abstract class BaseTemplateUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		upgradeDDMTemplates();
+		upgradeFragmentEntries();
+	}
+
+	protected abstract Pattern getTemplatePattern() throws Exception;
+
+	protected abstract String getTemplatePatternReplacement() throws Exception;
+
+	protected void upgradeDDMTemplates() throws Exception {
```

Comment:
@alberttgomes, at this moment, I believe that there is no need to use `protected` in the methods `upgradeDDMTemplates` and `upgradeFragmentEntries`, after all they won't be modified by the classes extending the `BaseTemplateUpgradeProcess`. Therefore, can you please change the access modifiers of these two methods to be `private`?